tests/check_framework/test_security.py

```diff
@@ -464,3 +467,93 @@ def test_with_referrer_policy(self):
     )
     def test_with_invalid_referrer_policy(self):
         self.assertEqual(base.check_referrer_policy(None), [base.E023])
+
+
+class CheckCacheLocationTest(SimpleTestCase):
+    @override_settings(
+        CACHES={
+            'default': {
+                'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
+                'LOCATION': Path.cwd() / 'tmp',
+            }
+        },
+        MEDIA_ROOT=Path.cwd(),
+    )
+    def test_exposed_on_media_root_path(self):
+        self.assertEqual(base.check_cache_and_media_root(None), [Warning(
+            "Your cache configuration might expose or corrupt your file system.",
+            hint="Your cache 'default' is inside your 'MEDIA_ROOT'",
+            id='security.W024',
+        )])
+
+    @override_settings(
+        CACHES={
+            'default': {
+                'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
+                'LOCATION': Path.cwd(),
+            }
+        },
+        MEDIA_ROOT=Path.cwd() / 'tmp',
+    )
+    def test_no_exposed_on_media_root_path(self):
+        self.assertEqual(base.check_cache_and_media_root(None), [])
```

Comment:
Hmm. This seems funny to me. Yes, we shouldn't allow the cache to be exposed in the media/static paths, but I also think that we shouldn't allow the media/static paths to come under the cache path either.

Reply:
oh I'll change it😅

Reply:
Oh it just for readable, I'll change it😅

Reply:
Ok, but I am also saying that we should also warn if the media/static resources are underneath the cache path.

It should be a simple case of also checking the inverse of the condition and wording the hint the opposite way around.

Reply:
It seems like I should add the check code under the follow🤔

```python
        if any(path in cache_path.parents for path in paths):
            errors.append(Warning(
                "Your cache configuration might expose or corrupt your file system.",
                hint=f"Your cache '{alias}' is inside your '{name}'",
                id='cache.W002',
            ))
       if any(cache_path in path.parents for path in paths):
            errors.append(Warning("..."))
```